docs/manual/build_system.qbk

```diff
@@ -6,44 +6,105 @@
+
+* [*The __hpx__ core libraries]: This forms the basic set of __hpx__ libraries. The generated targets are:
+  * hpx: The core __hpx__ library (always enabled).
+  * hpx_init: The __hpx__ initialization library that applications need to link
+    against to define the __hpx__ entry points (disabled for static builds).
+  * iostreams_component: The component used for (distributed) IO (always enabled).
+  * component_storage_component: The component needed for migration to persistant storage.
+  * unordered_component: The component needed for a distributed hash table.
+* [*__hpx__ Examples]: This target is enabled by default and builds all __hpx__ examples
+  (disable by setting HPX_WITH_BUILD_EXAMPLES=Off). __hpx__ examples are part of the
+  'all' target and are included in the installation if enabled.
+* [*__hpx__ Tests]: This target builds the __hpx__ test suite and is enabled by default
+  (disable by setting HPX_WITH_BUILD_TESTS=Off). They are not built by the 'all'
+  target and have to be built separately.
+* [*__hpx__ Documentation]: This target builds the Documentation, this is not
+  enabled by default. For more information see
+  [link hpx.manual.build_system.building_hpx.doc_tools Setting up the __hpx__ Documentation Tool Chain].
+
+For a complete list of available __cmake__ variables that influence the build of
+__hpx__ see [link hpx.manual.build_system.building_hpx.cmake_variables CMake Variables used to configure __hpx__].
+
+The variables can be used to refine the recipes that can be found
+[link hpx.manual.build_system.building_hpx.build_recipes here] which show some basic steps on
+how to build HPX for a specific platform
+
+In order to use __hpx__, only the core libraries are required (the ones marked
+as optional above are truly optional). When building against __hpx__, the __cmake__
+variable `HPX_LIBRARIES` will contain hpx and hpx_init (for pkgconfig, those are
+added to the `Libs` sections). In order to use the optional libraries, you need
+to specify them as link dependencies in your build.
+
+As __hpx__ is a modern C++ Library we require a certain minimal set of features
+from the C++11 standard. In addition, we make use of certain C++14 features if
+the used compiler supports them. This means that the __hpx__ build system will
+try to determine the highest support C++ standard flavor and check for availability
+of those features. If you want to force __hpx__ to use a specific C++ standard
+version you can use the following __cmake__ variables:
+
+* [*`HPX_WITH_CXX0X`]: Enables Pre-C++11 support (This is the minimal required
+    mode on older gcc versions).
+* [*`HPX_WITH_CXX11`]: Enables C++11 support
+* [*`HPX_WITH_CXX14`]: Enables C++14 support
+* [*`HPX_WITH_CXX0Y`]: Enables (experimental) C++17 support
+
+[heading:build_types Build Types]
+
+__cmake__ can be configured to generate project files suitable that enabled
+debugging support or for an optimized build. The __cmake__ variable used to set
```

Comment:
Did you mean: 

    __cmake__ can be configured to generate project files suitable for builds that have 
    enabled debugging support or for an optimized build (without debugging support)

(or similar)?